src/int.rs

```diff
@@ -0,0 +1,470 @@
+    pub const fn as_limbs(&self) -> &[Limb; LIMBS] {
+        self.0.as_limbs()
+    }
+
+    /// Borrow the limbs of this [`Int`] mutably.
+    pub fn as_limbs_mut(&mut self) -> &mut [Limb; LIMBS] {
+        self.0.as_limbs_mut()
+    }
+
+    /// Convert this [`Int`] into its inner limbs.
+    pub const fn to_limbs(self) -> [Limb; LIMBS] {
+        self.0.to_limbs()
+    }
+
+    /// Convert to a [`NonZero<Int<LIMBS>>`].
+    ///
+    /// Returns some if the original value is non-zero, and false otherwise.
+    pub const fn to_nz(self) -> ConstCtOption<NonZero<Self>> {
+        ConstCtOption::new(NonZero(self), self.0.is_nonzero())
+    }
+
+    /// Convert to a [`Odd<Int<LIMBS>>`].
+    ///
+    /// Returns some if the original value is odd, and false otherwise.
+    pub const fn to_odd(self) -> ConstCtOption<Odd<Self>> {
+        ConstCtOption::new(Odd(self), self.0.is_odd())
+    }
+
+    /// Get the sign bit of this [`Int`] as `Choice`.
+    pub fn sign_bit(&self) -> Choice {
+        ConstChoice::from_word_msb(self.0.to_words()[LIMBS - 1]).into()
+    }
+
+    /// View the data in this type as an [`Uint`] instead.
+    pub const fn as_uint(&self) -> Uint<LIMBS> {
+        self.0
+    }
+
+    /// Whether this [`Int`] is equal to `Self::MIN`.
+    pub fn is_minimal(&self) -> Choice {
+        Choice::from((self == &Self::MIN) as u8)
+    }
+
+    /// Whether this [`Int`] is equal to `Self::MAX`.
+    pub fn is_maximal(&self) -> Choice {
+        Choice::from((self == &Self::MAX) as u8)
+    }
+
+    /// Perform the two's complement "negate" operation on this [`Int`]:
+    /// map `self` to `(self ^ 1111...1111) + 0000...0001`
+    ///
+    /// Returns
+    /// - the result, as well as
+    /// - whether the addition overflowed (indicating `self` is zero).
+    ///
+    /// Warning: this operation is unsafe; when `self == Self::MIN`, the negation fails.
+    #[inline]
+    pub const fn negate_unsafe(&self) -> (Self, ConstChoice) {
+        let (val, carry) = self.0.wrapping_negc();
+        (Self(val), ConstChoice::from_word_lsb(carry))
+    }
+
+    /// Perform the [two's complement "negate" operation](Int::negate_unsafe) on this [`Int`]
+    /// if `negate` is truthy.
+    ///
+    /// Returns
+    /// - the result, as well as
+    /// - whether the addition overflowed (indicating `self` is zero).
+    ///
+    /// Warning: this operation is unsafe; when `self == Self::MIN` and `negate` is truthy,
+    /// the negation fails.
+    #[inline]
+    fn negate_if_unsafe(&self, negate: Choice) -> (Int<LIMBS>, Choice) {
+        let (negated, is_zero) = self.negate_unsafe();
+        (
+            Self(Uint::ct_select(&self.0, &negated.0, negate)),
+            is_zero.into(),
+        )
+    }
+
+    /// Map this [`Int`] to `-self` if possible.
+    ///
+    /// Yields `None` when `self == Self::MIN`, since an [`Int`] cannot represent the positive
+    /// equivalent of that.
+    pub fn neg(&self) -> CtOption<Self> {
```

Comment:
move it from here to neg.rs (all other negation related methods). Similarly to what Uint has.

Reply:
done!